hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rolloutgrouptargets/RolloutGroupTargetsBeanQuery.java

```diff
@@ -138,18 +138,19 @@ protected void saveBeans(final List<ProxyTarget> arg0, final List<ProxyTarget> a
     public int size() {
         long size = 0;
 
-        if (rolloutGroup.isPresent()) {
-            try {
-                firstPageTargetSets = getRolloutGroupManagement().findAllTargetsWithActionStatus(
-                        new PageRequest(0, SPUIDefinitions.PAGE_SIZE, sort), rolloutGroup.get().getId());
-                size = firstPageTargetSets.getTotalElements();
-            } catch (final EntityNotFoundException e) {
-                LOG.error("Rollout does not exists. Redirect to Rollouts overview", e);
-                rolloutUIState.setShowRolloutGroupTargets(false);
-                rolloutUIState.setShowRollOuts(true);
-                return 0;
-            }
+        try {
+            firstPageTargetSets = rolloutGroup.map(group -> getRolloutGroupManagement()
```

Comment:
Yes, this is a performance optimisation as it allows the loadBeans method to skip the query.